bgpd/bgp_route.c

```diff
@@ -1844,14 +1844,17 @@ int subgroup_announce_check(struct bgp_node *rn, struct bgp_path_info *pi,
 	 */
 	if (NEXTHOP_IS_V6) {
 		attr->mp_nexthop_len = BGP_ATTR_NHLEN_IPV6_GLOBAL;
-		if ((CHECK_FLAG(peer->af_flags[afi][safi],
-				PEER_FLAG_NEXTHOP_LOCAL_UNCHANGED)
-		     && IN6_IS_ADDR_LINKLOCAL(&attr->mp_nexthop_local))
-		    || (!reflect
-			&& IN6_IS_ADDR_LINKLOCAL(&peer->nexthop.v6_local)
-			&& peer->shared_network
-			&& (from == bgp->peer_self
-			    || peer->sort == BGP_PEER_EBGP))) {
+		if (((CHECK_FLAG(peer->af_flags[afi][safi],
```

Comment:
this "if" statement is getting a bit long and hard to understand... might it be worth thinking about how to break this apart in some way ... ??

Reply:
I will try to simplify the check and upload the changes.